specification/under_development/Compilation_And_Targeting.md

```diff
@@ -0,0 +1,67 @@
+A high degree of flexibility and configurability is hence of great importance when conceptualizing different compilation stages and how front- and backends integrate with the QIR toolchain and ecosystem.
+At the same time, a clear understanding of the purpose and functionality of subsequent stages in the compilation process is absolutely vital to grow a robust and performant compiler infrastructure over time.
+
+In addition to the [design goals of QIR](https://github.com/qir-alliance/qir-spec/blob/main/Scope.md) itself, we hence define the following goals for the compiler infrastructure around it:
+
+- Unify the ecosystem when reasonable; <br/>
+  Certain design choices can easily and often unintentionally lead to a fragmentation of the ecosystem of quantum application, libraries, and compilation tools, making it harder to leverage existing work in a new context. For example, requiring application code to be expressed in terms of a backend specific set of quantum instructions impedes portability of that code and hampers the development of shared application libraries.
+
+- Allow to make use of backend specific instructions at the application level; <br /> 
+  Exposing access to backend specific instructions is a valuable tool for research purposes. While we encourage libraries to be built against a hardware agnostic default set of instructions ([Default QIS]()), we aim for a simple mechanism for deviating from that default for the purpose of experimenting with new hardware intrinsics and developing backend specific compiler optimizations. These development in turn inform what abstractions are beneficial to expose in the Default QIS, and permit to ultimately incorporate fruitful optimizations into a target specific compilation stage.
+  
+- Define (and follow clear) guideance for defining backend specific instruction sets; <br />
+  ...
+
+TODO: the same as what is mentioned under bullet 1 applies to profiles
+
+## Compilation Stages
+
+- Language specific compilation: Compilation of source code to QIR bitcode
+- General optimizations: Optimizations that map QIR -> QIR and that do not limit compatibility in any way
+- General linking: Combines bitcode from different sources and libraries according to [standard linking practices]()
+- QIS resolution: Links a library (bitcode) taking naming conventions into account and giving strict precedence to the definitions in that library over definitions in the source code, unless otherwise indicated by a function attribute, and resolves precision if needed.
```

Comment:
I do not understand this. Is this to be understood as that there is a "library" which has the effect that if I link against it and there is duplicate symbols, the library takes precedence and the symbol in my code is deleted, or how is this to be understood?

Reply:
I need to flesh out the description - these are more or just my notes for now. I need to process one doc after another. Let's first close on Base_Profile.md and then proceed to the next document.